### PR TITLE
'new version for search'

### PR DESCRIPTION
**Pull request by emma0925**
Implemented Search with keyword, search with time, and search the newest message, tests are not ye done

### CODE REVIEW

**Comment by 343719654** [comment]
By the way, the 343719654 account is also mine, I don't know why my intelliJ was using this account

**Review by parmism — APPROVED** [review]
Looks good! Please make sure to package the files and also import those packagase before merging your code.

File: src/main/java/use_cases/conversation_search_use_case/chat/SearchInputBoundary.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import java.util.ArrayList;
+
+
+/**
+ * The input boundary for the login use case.
+ */
+// Note: The interface that the SearchUseCase implements
+// It specifies what the input (arguments) and output (return type) are.
+
+public interface SearchInputBoundary {
```
Comment (milkdromeda) [inline src/main/java/use_cases/conversation_search_use_case/chat/SearchInputBoundary.java:13]: It's good that you have an input boundary, which follows clean architecture.

**Review by milkdromeda — DISMISSED** [review]
Hi Emma, your code looks good.

I already communicated the following comment verbally (and you made the change), here it is again for reference:

For getMsgContent under use cases' keyword search: Maybe you can search a message only if the type is a TextMessage? Here you use getMsgContent, but the method does not make as much sense for non-text messages. Using my suggestion, you would not need an abstract getMsgContent method.

**Review by amyyong11 — APPROVED** [review]
I think the only thing you may need is the UI, but everything else looks good!

**Comment by emma0925** [comment]
> 

Yes, just repackaged yesterday. Thank you to bring this up so I could fix it.

**Comment by emma0925** [comment]
> I think the only thing you may need is the UI, but everything else looks good!

Right, I am going to create the UI for it tomorrow.

**Review by parmism — APPROVED** [review]
Thank you!

**Review by parmism — DISMISSED** [review]

**Review by amyyong11 — DISMISSED** [review]

**Review by milkdromeda — DISMISSED** [review]

**Pull request merged**